addons/pos_self_order/models/product_product.py

```diff
@@ -71,6 +71,8 @@ def _load_pos_self_data(self, data):
             order='sequence,default_code,name',
             load=False
         )
+        for product in products:
+            product['image_128'] = bool(product['image_128'])
```

Comment:
Why not remove `image_128` from `fields` in a first place? If I'm correct, this will load all images, then replace them by a boolean.

Reply:
@nim-odoo It is used in getImageUrl to check if the product has an associated image to load. If no image is available, it adjusts the view.

Reply:
Then it might make sense to introduce a `has_image` field instead of relying on this hack in several places, no?

Reply:
@nim-odoo you are right. I'll inform PoS RD to do it in master.